### PR TITLE
Adds a drone dispenser to Boxstation

### LINKED ISSUE #12282: Revenants can become drones
Click on it and it gives you the prompt to become one

### PR DESCRIPTION
**Pull request by ghost**
The dispenser is in a formerly empty space next to an oxygen closet. A small light bulb has also been added next to it, for visibility.

In addition, whenever a drone shell is created, it displays ghosts with a prompt to enter

### CODE REVIEW

**Comment by JJRcop** [comment]
Definitely confused as to why revenants can call attack_ghost...

**Comment by Anonus** [comment]
That doesn't totally fix that issue unless posibrains and swarmers also check it.

**Comment by JJRcop** [comment]
i found the issue, you should fix this instead of doing what you're doing

https://github.com/tgstation/-tg-station/commit/e992601abf6a916dccd7ccd0c3724584f390d7ba#commitcomment-13730091

**Comment by Shadowlight213** [comment]
:+1:

**Comment by Zilenan91** [comment]
Can you make it so that if it's emagged it shits out hacked drones

**Comment by JJRcop** [comment]
Really having revenants able to attack_ghost is bad, fix it properly.

**Comment by Anonus** [comment]
It should really just call `examine` they don't need any of what `attack_ghost` does

**Comment by Fox-McCloud** [comment]
Rogue mech/robot fab when?

=p

**Comment by Dorsisdwarf** [comment]
If you're doing this, remove them from robotics. Therefore you have the tradeoff that while the ventcrawl superspeed engineers are available every round from roundstart, if you cut off the supply, then you cut off the supply.

File: code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm [hunk 1]
```diff
@@ -16,16 +16,28 @@
...
+		user << "<span class='warning'>This drone shell has already been activated.</span>"
 	var/mob/living/simple_animal/drone/D = new drone_type(get_turf(loc))
 	D.key = user.key
 	qdel(src)
 
+/obj/item/drone_shell/New()
+	notify_ghosts("An unactivated [name] has been created in [get_area(src)]. <a href=?src=\ref[src];ghostjoin=1>(Click to enter)</a>")
+	..()
+
+/obj/item/drone_shell/Topic(href, href_list)
```
Comment (Giacom) [inline code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm]: Make sure to call ..() in Topic().

**Comment by Xhuis** [comment]
@Dorsisdwarf Drone dispensers start off with little materials and are only good for a few Drone shells unless reupplied. For this reason, I'm leaving them available from exosuit fabricators.

**Comment by Jalleo** [comment]
Tbh that is a awful place to put the dispensor.

**Comment by Xhuis** [comment]
@Jalleo Where do you think it should go?

**Comment by Shadowlight213** [comment]
@Jalleo it's a fantastic place. I've placed it there many times and it works well. It's in maint, meaning it's easy to destroy unseen and is easy for ops to take out if they are bombing comms without needing to go out of their way. Being sort of near engineering means that drones can grab materials and rcds relatively quickly from there, but it is far enough away to have a chance to survive a singuloose. It also fits in there well and doesn't stick out.

**Comment by Jalleo** [comment]
@Xhuis electrical maint it makes sense thematically and puts it out of the way from tcomm bombing.
Bombing tcomms then losing as ops shouldn't mean the drone fab is gone.

**Comment by Bawhoppen** [comment]
I think it should go in the construction area IMO

**Comment by Shadowlight213** [comment]
@Jalleo I don't like that spot as it means drones would have to run all the way across the station to get to engineering for materials and rcds

**Comment by Dorsisdwarf** [comment]
drone machine should go kaput if the snydies bomb tcomms

**Pull request closed without merging**